types/wordpress__blocks/api/registration.d.ts

```diff
@@ -117,11 +117,15 @@ export function registerBlockStyle(blockName: string, styleVariation: BlockStyle
  * behavior. Once registered, the block is made available as an option to any
  * editor interface where blocks are implemented.
  *
- * @param name - Block name.
+ * @param blockNameOrMetadata - Block type name or its metadata.
  * @param settings - Block settings.
  *
  * @returns The block if it has been successfully registered, otherwise `undefined`.
  */
+export function registerBlockType<T extends Record<string, any> = {}>(
+    metadata: BlockConfiguration<T>,
+    settings?: BlockConfiguration<T>,
+): Block<T> | undefined;
```

Comment:
I see we already use `T` as a generic type parameter, but I feel like we can do better than that and help people understand the purpose more clearly if we choose a name related to its use in the block. maybe something like `Attributes` or `BlockAttributes`?

cc: @sirreal @sarayourfriend have we discussed naming of type parameters?

Reply:
Proposing `TAttributes` here because from the context it should already clear that the block attributes are meant.
E.g. in React components we also write `ButtonProps` instead of `ButtonComponentProps`.

Reply:
`Attributes` or `TAttributes` seem fine. I'm no fan of the `T` prefix but I understand there's a lot of carry-over from older times when string-typing was more common.